cpp/sophus/sensor/orthographic.h

```diff
@@ -9,162 +9,44 @@
-    return scaleY() * (image_size_.height - TScalar(0.5)) + offsetX();
-  }
-
-  [[nodiscard]] OrthographicModelT subsampleDown() const {
-    FARM_FATAL("not implemented");
-  }
-
-  [[nodiscard]] OrthographicModelT subsampleUp() const {
-    FARM_FATAL("not implemented");
-  }
-
-  [[nodiscard]] OrthographicModelT binDown() const {
-    FARM_FATAL("not implemented");
-  }
-
-  [[nodiscard]] OrthographicModelT binUp() const {
-    FARM_FATAL("not implemented");
-  }
-
-  [[nodiscard]] DistorationParams distortionParams() const {
-    return params_.template tail<kNumDistortionParams>();
-  }
-
-  Eigen::Matrix<TScalar, kNumParams, 1>& params() { return params_; }
-
-  [[nodiscard]] Eigen::Matrix<TScalar, kNumParams, 1> const& params() const {
-    return params_;
-  }
-
-  [[nodiscard]] PixelImage camProj(PointCamera const& point_camera) const {
-    return Proj::template camProj(params_, point_camera);
-  }
-
-  [[nodiscard]] PointCamera camUnproj(
-      PixelImage const& pixel_image, double depth_z) const {
-    return Proj::template camUnproj(params_, pixel_image, depth_z);
-  }
-
-  TScalar* data() { return params_.data(); }
-
-  [[nodiscard]] ImageSize const& imageSize() const { return image_size_; }
-
-  /// Creates default pinhole model from `image_size`.
-  [[nodiscard]] bool contains(
-      Eigen::Vector2i const& obs, int border = 0) const {
-    return this->image_size_.contains(obs, border);
-  }
-
-  /// Creates default pinhole model from `image_size`.
-  [[nodiscard]] bool contains(
-      PixelImage const& obs, TScalar border = TScalar(0)) const {
-    return this->image_size_.contains(obs, border);
-  }
-
- private:
-  OrthographicModelT() = default;
-  ImageSize image_size_;
-  Eigen::Matrix<TScalar, kNumParams, 1> params_;
-};
-
-using OrthographicModel = OrthographicModelT<double>;
+using OrthographicModelT =
+    CameraModelT<TScalar, AffineTransform, ProjectionOrtho>;
 
 /// Returns orthographic camera model given bounding box and image size.
 template <class TScalar>
 OrthographicModelT<TScalar> orthoCamFromBoundingBox(
     Eigen::AlignedBox<TScalar, 2> const& bounding_box, ImageSize image_size) {
-  TScalar min_x = bounding_box.min().x();
-  TScalar min_y = bounding_box.min().y();
-  TScalar max_x = bounding_box.max().x();
-  TScalar max_y = bounding_box.max().y();
-
   // (-0.5, -0.5)   -> (min.x, min.y)
   // (-0.5, h-0.5)  -> (min.x, max.y)
   // (w-0.5, -0.5)  -> (max.x, min.y)
   // (w-0.5, h-0.5) -> (max.x, max.y)
   //
   // Thus we have the following relationship:
+  // u = x*sx + ox
+  // v = y*sy + oy
   //
-  // u*sx + ox = x
-  // v*sy + oy = y
+  // -0.5 = min.x * sx  + ox   => ox = -(0.5 + min.x * sx)
+  // (w-0.5) = max.x * sx + ox
   //
-  // -0.5*sx  + ox = min.x   => ox = min.x + 0.5*sx
-  // (w-0.5)*sx + ox = max.x
-  //
-  // (w-0.5)*sx + min.x + 0.5*sx = max.x  =>  sx = (max.x-min.x) / w
+  // w = sx * (max.x  - min.x)
+  // sx = w / (max.x  - min.x)
+
+  Eigen::Array<TScalar, 2, 1> const range =
```

Comment:
@stevenlovegrove did we have new types here? Maybe we should move them to sophus?